pyiron_base/job/core.py

```diff
@@ -739,18 +739,33 @@ def copy_to(self, project, new_job_name=None, new_database_entry=True, copy_file
 
         Args:
             project (JobCore/ProjectHDFio/Project): project to copy the job to
-            new_database_entry (bool): [True/False] to create a new database entry - default True
+            input_only (bool): [True/False] Whether to copy only the input. (Default is False.)
+            new_database_entry (bool): [True/False] Whether to create a new database entry. If input_only is True then
+                new_database_entry is False. (Default is True.)
             copy_files (bool): [True/False] copy the files inside the working directory - default True
 
         Returns:
             JobCore: JobCore object pointing to the new location.
         """
-        new_job_core, _, _, _ = self._internal_copy_to(
+        # Update flags
+        if input_only and new_database_entry:
+            new_database_entry = False
+
+        new_job_core, _, _, reload = self._internal_copy_to(
             project=project,
             new_job_name=new_job_name,
             new_database_entry=new_database_entry,
             copy_files=copy_files
         )
+        if reload_flag:
+            return new_job_core
+
+        # Remove output if it should not be copied
+        if input_only:
+            for group in new_job_core.project_hdf5.list_groups():
+                if "output" in group:
+                    del new_job_core.project_hdf5[posixpath.join(new_job_core.project_hdf5.h5_path, group)]
+            new_job_core.status.initialized = True
```

Comment:
@raynol-dsouza will this annoy us? @pmrv isn't the job still `created` and not `initialized`? I am not confident because I still don't grok the status flow, I just want to throw a yellow flag.

Reply:
@liamhuber I think it shouldn't cause any issues, since the existing version also set the status to `initialized`.

Reply:
This shouldn't change functionality, as this is the same as before it just lives on `JobCore` now instead of `GenericJob`.  I don't get the status hierarchy as well, but it's been on my backlog for some archeology.

Reply:
I looked it up and `initialized` comes before `created`, so I suppose we should be good.

Reply:
> I think it shouldn't cause any issues, since the existing version also set the status to initialized.

And I thought the existing behaviour was a pain in the butt for us 😉 

Anyhow, maintaining the existing behaviour is totally fine for now.

> but it's been on my backlog for some archeology.

I like this phrasing 😄

Reply:
> And I thought the existing behaviour was a pain in the butt for us 😉

@liamhuber The thing that was problematic for us, was that the reference job itself needed to be initialized, not it's copy. Creating a job and saving it without running, sets the status to `created`, and not `initialized`. The funny part is that while making a copy of the reference job, the existing code sets the copy's status to `initialized`, while the reference job still remains `created`. 😄 I do not know if this needs fixing.